src/plugins/data_source/server/legacy/client_config.ts

```diff
@@ -55,7 +55,7 @@ export function parseClientOptions(
       config.ssl?.certificateAuthorities
     );
 
-    sslConfig.ca = certificateAuthorities || [];
+    sslConfig.ca = certificateAuthorities;
```

Comment:
we also need to fix the interface to make ca as optional I think, at line 16

Reply:
That should already be covered because DataSourceSSLConfigOptions is a [Partial](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype)